n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+
+public class NQueenProblem { 
+    private static void placeQueenOnBoard(int Qi, int[] board) {
+        int n = board.length;
+        //base case
+        if (Qi == n) {
+            System.out.println(Arrays.toString(board));
```

Comment:
@Monal5031: `board` is an array that is passed as a parameter. Direct print in the main function should do. Returning status may be necessary.